File: Elixir/greeter/lib/greeter.ex
defmodule Greeter do

  def formattedName(name) do
    String.capitalize(String.trim(name))
  end

  def greet(name) do
    "Hello " <> formattedName(name)
  end

  def greet(name, time) when time.hour >= 6 and time.hour <= 12 do
    "Good morning " <> formattedName(name)
  end

  def greet(name, time) when time.hour >= 18 and time.hour <= 21 do
    "Good evening " <> formattedName(name)
  end

  def greet(name, time) when time.hour >= 22 or time.hour <= 5 do
    "Good night " <> formattedName(name)
  end
end

File: Kotlin/greeter/src/main/kotlin/Greeter.kt
import java.time.LocalDateTime
import java.util.*

class Greeter {
    fun greet(name: String): String {
        return greet(name, LocalDateTime.now())
    }

    fun greet(name: String, time: LocalDateTime): String {
        var prefix = "Hello"

        when (time.hour) {
            in 6..11 -> prefix = "Good morning"
            in 18..21 -> prefix = "Good evening"
            in 22..24, in 0..6 -> prefix = "Good night"
        }

        return "$prefix ${format(name)}"
    }

    private fun format(name: String) = name.trim()
        .replaceFirstChar {
            if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString()
        }
}